atlas/packages/app/src/components/Navbar/Navbar.tsx

```diff
@@ -1,60 +1,87 @@
 import React, { useState } from 'react'
-import { navigate, Link, RouteComponentProps } from '@reach/router'
+import { RouteComponentProps, Link, navigate } from '@reach/router'
 
-import routes from '@/config/routes'
 import { Icon } from '@/shared/components'
-import { Header, NavigationContainer, StyledIcon, StyledSearchbar, CancelButton, Logo } from './Navbar.style'
+import routes from '@/config/routes'
+import {
+  Header,
+  NavigationContainer,
+  StyledIcon,
+  StyledSearchbar,
+  SearchbarContainer,
+  CancelButton,
+  CancelButtonContainer,
+  Logo,
+} from './Navbar.style'
 
-const Navbar: React.FC<RouteComponentProps> = () => {
-  const [search, setSearch] = useState('')
-  const [isSearching, setIsSearching] = useState(false)
+type NavbarProps = RouteComponentProps
 
-  const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
-    setSearch(e.currentTarget.value)
-  }
+const Navbar: React.FC<NavbarProps> = () => {
+  const [search, setSearch] = useState('')
+  const [focus, setFocus] = useState(false)
 
   const handleKeyPress = (e: React.KeyboardEvent<HTMLInputElement>) => {
-    if (e.key === 'Enter' || (e.key === 'NumpadEnter' && search.trim() !== '')) {
+    if (e.key === 'Enter' || (e.key === 'NumpadEnter' && search.trim())) {
       navigate(routes.search(search))
     }
+    if (e.key === 'Escape' || e.key === 'Delete' || e.key === 'Esc') {
+      setFocus(false)
+      setSearch('')
+    }
+  }
+
+  const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    if (search !== '' && e.currentTarget.value === '') {
+      setFocus(false)
+    } else {
+      setFocus(true)
+    }
+    setSearch(e.currentTarget.value)
   }
 
   const handleFocus = () => {
-    setIsSearching(true)
+    setFocus(true)
   }
 
   const handleCancel = () => {
     setSearch('')
-    setIsSearching(false)
+    setFocus(false)
   }
   return (
-    <Header isSearching={isSearching}>
-      {!isSearching && (
-        <NavigationContainer>
-          <Link to="/">
-            <Logo />
-          </Link>
-          <Link to="/">
-            <StyledIcon name="home" />
-          </Link>
-          <Link to={routes.browse()}>
-            <StyledIcon name="binocular" />
-          </Link>
-        </NavigationContainer>
-      )}
-
-      <StyledSearchbar
-        placeholder="Search..."
-        onChange={handleChange}
-        value={search}
-        onKeyPress={handleKeyPress}
-        onFocus={handleFocus}
-      />
-      {isSearching && (
-        <CancelButton onClick={handleCancel}>
-          <Icon name="times" />
-        </CancelButton>
-      )}
+    <Header hasFocus={focus}>
+      <div>
```

Comment:
Do we need that `div` here? It will be empty when `focus` is `true`

Reply:
I've added those divs to make animating the `Searchbar` easier, they allow me to keep the same grid structure even when one element is missing, having the elements themselves instead of enclosing them in divs makes the `Searchbar`  jump around (it is the 2nd column when unfocused, 1st when focused). An alternative was to hide the `CancelButton` (or the `NavButtons` respectively) when `Searchbar` is unfocused, but that looked to me like a worse solution.

Reply:
Okay, let's keep it